dhcpv4/nclient4/example_lease_test.go

```diff
@@ -0,0 +1,65 @@
+//this is an example for nclient4 with lease/release
+
+package nclient4_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv4/nclient4"
+	"github.com/vishvananda/netlink"
+)
+
+//applyLease adding the assigned ip to the interface specified by ifname
+func applyLease(lease *nclient4.Lease, ifname string) error {
+	link, err := netlink.LinkByName(ifname)
+	if err != nil {
+		return err
+	}
+	prefixlen := 32
+	if ipmask := lease.ACK.SubnetMask(); ipmask != nil {
+		prefixlen, _ = ipmask.Size()
+
+	}
+	prefixstr := fmt.Sprintf("%v/%v", lease.ACK.YourIPAddr, prefixlen)
+	naddr, err := netlink.ParseAddr(prefixstr)
+	if err != nil {
+		return err
+	}
+	err = netlink.AddrReplace(link, naddr)
+	return err
+
+}
+
+func main() {
+	ifname := "eth1.200"
+	remoteid := "client-1"
+	var idoptlist dhcpv4.OptionCodeList
+	//specify option82 is part of client identification used by DHCPv4 server
+	idoptlist.Add(dhcpv4.OptionRelayAgentInformation)
+	clntOptions := []nclient4.ClientOpt{nclient4.WithClientIDOptions(idoptlist), nclient4.WithDebugLogger()}
+	clnt, err := nclient4.New(ifname, clntOptions...)
+	if err != nil {
+		log.Fatalf("failed to create dhcpv4 client,%v", err)
+	}
+	//adding option82/remote-id option to discovery and request
+	remoteidsubopt := dhcpv4.OptGeneric(dhcpv4.AgentRemoteIDSubOption, []byte(remoteid))
+	option82 := dhcpv4.OptRelayAgentInfo(remoteidsubopt)
+	_, lease, err := clnt.Request(context.Background(), dhcpv4.WithOption(option82))
+	if err != nil {
+		log.Fatal(err)
+	}
+	//print the lease
+	log.Printf("Got lease:\n%+v", lease)
+	//apply the lease
+	applyLease(lease, ifname)
```

Comment:
the code is looking great to me, but I'm not sure how we're gonna run this test automatically. This seems like it'll only work if we run this test somewhere where we're root and we have an unconfigured interface? That's pretty unlikely to run on something like CircleCI or Travis

Reply:
since now I have decided no longer mandate user to add assigned address to the interface, and Release() use Client.conn directly, I could try to write some test cases using socketpair, just like the cases in client_test.go;